src/clickonce/native/projects/NetCoreCheck/EXE/NetCoreCheckExe.cpp

```diff
@@ -4,49 +4,78 @@
 
 #include "NetCoreCheckExe.h"
 
+#define PARAM_NAME(s) (_wcsicmp(*argv, TEXT(s)) == 0)
+
 // Globals
 Logger *g_log;
+const char* g_help = "NETCoreCheck [options]\n"
+"-n, --runtimename -       Runtime name                    (Example: Microsoft.AspNetCore.App)\n"
+"-v, --runtimeversion -    Runtime version in format x.y.z (Example: 1.2.3)\n"
+"-r, --rollforwardpolicy - (Optional) Roll forward policy  (Example: LatestMajor)\n"
+"-c, --runtimeconfigfile - Path to runtime config file     (Example: c:\\Foo\\Bar.runtimeconfig.json)\n"
+"-l, --logfile -           (Optional) Path to log file\n\n"
+"If runtimeconfigfile is specified then runtimename, runtimeversion and rollforwardpolicy shouldn't be.\n\n"
+"If 0 is returned the runtime requirement is satisfied.\n\n"
+"Examples:\n\n"
+"NETCorecheck --runtimename Microsoft.AspNetCore.App --runtimeversion 3.1.0\n"
+"NETCorecheck -n Microsoft.WindowsDesktop.App -v 5.0.1 -r LatestMajor\n"
+"NETCorecheck -c c:\\Foo\\Bar.runtimeconfig.json -l c:\\Foo\\Bar.log\n";
 
 int __cdecl wmain(int argc, WCHAR* argv[])
 {
-    if (argc < 3 || argc > 5)
+    LPCWSTR runtimeName = NULL;
+    LPCWSTR runtimeversion = NULL;
+    LPCWSTR rollForwardPolicy = NULL;
+    LPCWSTR existingRuntimeConfigFilPath = NULL;
+    LPCWSTR logFilePath = NULL;
+
+    // Parse the command line options
+    argv++;
+    for (int i = 1; i < argc; i++)
     {
-        return EXIT_FAILURE_INVALIDARGS;
+        if (*argv != nullptr)
+        {
+            if (PARAM_NAME("-n") || PARAM_NAME("--runtimename"))
+            {
+                runtimeName = *++argv;
```

Comment:
Would this fail for incorrectly formed command line, i.e. a switch without a value specified?

Reply:
Thanks for noticing, I now correctly detect that scenario and display help